tdesign/src/Rows/SQLCredentialsNicknameRow.tsx

```diff
@@ -1,22 +1,118 @@
-import { Box, Grid, Typography, FormControl, Button } from "@mui/material";
-import { InputWithCopy /*Input*/ } from "../Input";
-import { IconDelete } from "../Icon";
+import {
+  Box,
+  Grid,
+  Typography,
+  FormControl,
+  Button,
+  OutlinedInput,
+  InputAdornment,
+} from "@mui/material";
+import { styled } from "@mui/material/styles";
+import React from "react";
+import { IconCopy, IconDelete } from "../Icon";
 
 interface ISQLCredentialsNicknameRowProps {
   name?: string;
+  nickname?: string;
   handleDelete?: () => void;
-  idNicknamePrefix?: string;
   idUsernamePrefix?: string;
   deleteButton?: React.ReactNode;
+  saveNicknameButton?: React.ReactNode;
+  setNewNickname?: (nickname: string) => void;
+  onCopy?: () => void;
 }
 
+const UsernameField = styled(Box)(({ theme }) => ({
+  display: "flex",
+  backgroundColor: theme.palette.surfaces.sql.main,
+  borderRadius: "4px",
+  flexGrow: 1,
+  boxSizing: "border-box",
+  height: "100%",
+  flexDirection: "row",
+  alignContent: "center",
+  alignItems: "center",
+  padding: "4px 16px 4px 16px",
+  fontFamily: "monospace",
+}));
+
+const CopyIconBox = styled(Box)({
+  display: "flex",
+  flexGrow: 0,
+  ":hover": {
+    cursor: "pointer",
+  },
+});
+
+const DeleteIconBox = styled(Box)({
+  display: "flex",
+  flexGrow: 0,
+});
+
+const Wrapper = styled(Box)({
+  boxSizing: "border-box",
+  display: "flex",
+  flexDirection: "column",
+  height: "100%",
+});
+
+const Container = styled(Box)({
+  display: "flex",
+  boxSizing: "border-box",
+  flexDirection: "row",
+  flexGrow: 1,
+});
+
+const UsernameWithBreak = ({ username }: { username: string }) => {
+  const splitIndex = Math.floor(username.length / 2);
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        flexGrow: 1,
+      }}
+    >
+      {username.slice(0, splitIndex)}
+      <wbr />
+      {username.slice(splitIndex)}
+    </Box>
+  );
+};
+
 const SQLCredentialsNicknameRow = ({
   name,
   handleDelete,
-  // idNicknamePrefix,
   idUsernamePrefix,
-  deleteButton,
+  deleteButton = (
+    <Button onClick={handleDelete}>
+      <IconDelete />
+    </Button>
+  ),
+  nickname,
+  saveNicknameButton,
+  setNewNickname,
+  onCopy,
 }: ISQLCredentialsNicknameRowProps) => {
+  const codeRef = React.useRef<HTMLElement>(null);
+  const handleClickCopy = () => {
```

Comment:
This looks like it was copied (hah) from `PreWithCopy`. Any reason not to use that? (I think it's fine)

Reply:
It was! :)
I originally used `PreWithCopy` and turned to a div because of line breaking / alignment issues.